### ISSUE TITLE
Panel and Fullscreen display addons don't play together

### ISSUE THREAD

**Issue by marcguyer** [codemirror/CodeMirror#3144]
Since the panel addon adds a DOM node outside of the CodeMirror wrapper node, the panel is not part of the fullscreen mode. 

Perhaps the panel addon should be modified to prepend/append the panel node to the CodeMirror wrapper instead? The addon [creates another wrapper](https://github.com/codemirror/CodeMirror/commit/183ebcb75dbb9554ad6c14c47307fdc69d5ac45c#diff-7a00cfa89a65ce98069ccbefb219fe55R58) around the cm.getWrapperElement(), then adds the new panels to the new wrapper. That doesn't work with fullscreen.

Perhaps another option is for all display addons to share a common wrapper.

**Comment by marijnh** [comment]
Yes, this is a known issue. I think in the next major release I will add another wrapper element around the editor, to make it easier to size the editor and its content independently. That'd be rather invasive, with regards to the way the editor is styled with CSS, so it must wait until a backwards-incompatible release (6.0).

**Comment by oslyak** [comment]
My not very smart implementation with jQuery, but works. ;-)
Works only with top panels and panel should have classes: "cm-panel top"

    // codemirror/fullscreen.scss
    .CodeMirror-fullscreen {
      position: fixed;
      top: 0; left: 0; right: 0; bottom: 0;
      height: auto;
      z-index: 9;
      padding-top: 40px;
    }
    
    .cm-panel.top.cm-panel-fullscreen {
      position: fixed;
      z-index: 10;
      top: 0; left: 0;
      width: 100%;
    }

    // codemirror/fullscreen.js
    .....
      function setFullscreen(cm) {
        var wrap = cm.getWrapperElement();
        cm.state.fullScreenRestore = {scrollTop: window.pageYOffset, scrollLeft: window.pageXOffset,
                                      width: wrap.style.width, height: wrap.style.height};
        wrap.style.width = "";
        wrap.style.height = "auto";
        wrap.className += " CodeMirror-fullscreen";
        $(wrap).siblings(".cm-panel.top").addClass("cm-panel-fullscreen");   // This string added!
        document.documentElement.style.overflow = "hidden";
        cm.refresh();
      }
    
      function setNormal(cm) {
        var wrap = cm.getWrapperElement();
        wrap.className = wrap.className.replace(/\s*CodeMirror-fullscreen\b/, "");
        $(".cm-panel").removeClass("cm-panel-fullscreen");  // This string added!
        document.documentElement.style.overflow = "";
        var info = cm.state.fullScreenRestore;
        wrap.style.width = info.width; wrap.style.height = info.height;
        window.scrollTo(info.scrollLeft, info.scrollTop);
        cm.refresh();
      }

**Comment by Lakhman** [comment]
@oslyak Works great for me, tested on iPad and Mac, nice simple fix, 2 lines to JS and the CSS, awesome! Thanks!
